### ISSUE TITLE
Does Tds.Ecto support concurrent tests?

### ISSUE THREAD

**Issue by MichaelKorn** [livehelpnow/tds_ecto#76]
The documentation for Ecto.Adapters.SQL.Sandbox makes it very clear (https://hexdocs.pm/ecto_sql/Ecto.Adapters.SQL.Sandbox.html#module-database-support):

> While both PostgreSQL and MySQL support SQL Sandbox, only PostgreSQL supports concurrent tests while running the SQL Sandbox. Therefore, do not run concurrent tests with MySQL as you may run into deadlocks due to its transaction implementation.

The same should be done for Tds.Ecto. Since I got transaction deadlocks with Tds.Ecto, I am currently assuming that it is not supported.

**Comment by mjaric** [comment]
It is possible in new version I'm preparing in [ecto_sql ](https://github.com/elixir-ecto/ecto_sql/blob/master/lib/ecto/adapters/tds.ex#L114)(ecto 3). We are almost at the finish line and soon this will be out. 

And [here](https://github.com/elixir-ecto/ecto_sql/blob/master/integration_test/sql/transaction.exs#L57) is test as example that uses `:snapshot` isolation level 

You must configure database connection with `set_allow_snapshot_isolation: :on` and then use explicit transaction `Repo.transaction(fn -> ... end, isolation_level: :snapshot)`.
Basically this is same as calling `SET TRANSACTION ISOLATION LEVEL SNAPSHOT" within transaction.


In ecto < 3.0, there is similar thing, but it is "always on/off" and must be set in repo config. Flags are:

- :set_transaction_isolation_level - atom :read_uncommitted | :read_committed | :repeatable_read | :snapshot | :serializable
- :set_allow_snapshot_isolation - atom :on | :off

**Comment by MichaelKorn** [comment]
Thanks a lot @mjaric 
In the meantime it also works with ecto <3.0 after I set the necessary flags.
It took a while, because after the tests are async and running at the same time, there were a lot of non-deriministic effects. Many tests wait for messages or check that a wrong message is not coming. Unfortunately, the jitter in how long something in the tests takes is very high, now.
In addition, after changing the config, ecto.migrate failed first because "alter table" in isolation_level =: snapshot failed.

**Comment by mjaric** [comment]
Here is necessary setup for sandbox when using ecto_sql https://github.com/aaronrenner/tds_deadlock_demo/pull/1
